### PR TITLE
Fluent validation

### PR DESCRIPTION
**Pull request by pyotrritchie**
So i've used this [library](https://github.com/JeremySkinner/FluentValidation
) which allows us to functionally declare validation requirements on a resource objects properties. Appears to be the standard for .net web api's and works with both nancy and carter, it also required minimal changes on the client

### CODE REVIEW

File: tests/Integration/Service.Tests/TariffModuleSpecs/WhenAddingTariff.cs [hunk 1]
```diff
@@ -26,10 +26,10 @@ public void SetUp()
                 Description = "test-case"
             };
             var tariff = new Tariff
-                             {
-                                 TariffCode = "test",
-                                 Description = "test-case"
-                             };
+            {
+                TariffCode = "test",
+                Description = "test-case"
```
Comment (richardiphillips) [inline tests/Integration/Service.Tests/TariffModuleSpecs/WhenAddingTariff.cs]: I prefer the previous indentation personally

File: src/Service/Modules/TariffModule.cs [hunk 2]
```diff
@@ -62,10 +61,11 @@ private object AddTariff()
...
             resource.Links = new[] { new LinkResource("entered-by", employeeUri) };
-
-            var result = this.tariffService.Add(resource);
-
-            return this.Negotiate.WithModel(result);
+            var validator = new TariffValidator();
+            var results = validator.Validate(resource);
+            return validator.Validate(resource).IsValid
+                       ? this.Negotiate.WithModel(this.tariffService.Add(resource))
+                       : this.Negotiate.WithModel(results).WithStatusCode(HttpStatusCode.BadRequest);
```
Comment (richardiphillips) [inline src/Service/Modules/TariffModule.cs]: Do we need to change our error catching for this format?
Should we make a response processor for it?

File: tests/Integration/Service.Tests/TariffModuleSpecs/WhenAddingTariff.cs (hunk 1 above)
Comment (pyotrritchie) [reply tests/Integration/Service.Tests/TariffModuleSpecs/WhenAddingTariff.cs]: 👍

File: src/Service/Modules/TariffModule.cs (hunk 2 above)
Comment (pyotrritchie) [reply src/Service/Modules/TariffModule.cs]: Yes, I've changed how we handle the errors but only ever so slightly.. that commit is added now so you can see. I'd say that it's fine using the model that they build out, there isn't anything on the client that requires major changes... but then maybe there is something i've missed?

Comment (richardiphillips) [reply src/Service/Modules/TariffModule.cs]: No I think you're right

**Pull request merged**